In [2]:
import warnings

# Suppress all warnings
warnings.filterwarnings('ignore')

In [3]:
from IPython.display import display, HTML

import pandas as pd
import numpy as np

import matplotlib as plt 
import seaborn as sns

## 4.3 Overview of the explanatory variables

To make a good forecast model for energy consumption, it is important to understand the attributes that the variables have and know what each variable represents. Each variable serves to give a better understanding of what contributes to energy consumption for Andel Energi’s customers. By understanding the variables, a better understanding of energy usage patterns can be gained. Therefore, the following section will introduce and describe the variables contained within the dataset which has been provided by Andel Energi and other sources.

| Variable Name | Data Type | Variable Type | Description | Source | Additional Notes |
|---------------|-----------|---------------|-------------|--------|------------------|
| start_time | datetime | Numerical | Start time of the observation | Delivered by Andel | - |
| is_holiday | int | Binary | Indicates if the time is a holiday | python library *holidays* | 1 for holiday, 0 for non-holiday |
| month | str | Categorical | Month of the observation | python library *calender* | - |
| weekday | str | Categorical | Weekday of the observation | python library *calender* | - |
| EC_pct_change | float | Numerical | Percent change in electrical cars registrated | Gathered from DST.dk | - |
| HC_pct_change | float | Numerical | Percent change in plug-in hybrid cars registrated | Gathered from DST.dk | - |
| humidity_past1h | float | Numerical | Humidity level in the past hour | Gathered from DMI.dk | - |
| temp_mean_past1h | float | Numerical | Mean temperature in the past hour | Gathered from DMI.dk | - |
| wind_speed_past1h | float | Numerical | Wind speed in the past hour | Gathered from DMI.dk | - |
| EL_price | float | Numerical | Price of electricity | Gathered from DST.dk | - |
| sum_quantity | float | Numerical | Sum of energy consumption | Delivered by Andel | Dependent variable for analysis |

- **start_time & slut_time**: Timestamps marking the beginning and end of the recorded hour.
- **is_holiday**: A binary variable indicating whether the observation falls on a holiday.
- **month & weekday**: Time-related categorical variables aiding in temporal pattern identification.
- **EC_pct_change & HC_pct_change**: Percentage changes in energy consumption and heating, respectively, offering insights into variability and trends.
- **umidity_past1h, temp_mean_past1h, wind_speed_past1h**: Meteorological variables from the previous hour that might influence energy usage patterns.
- **EL_price**: Price of electricity, a potential determinant of consumption behavior.
- **sum_quantity**: Aggregated measure of energy consumption, serving as the dependent variable for analysis.

The foundation of any data analysis lies in understanding the basic statistical properties of a dataset. Summary statistics provide essential information that can guide initial conclusions and is a way to get directions for future analysis. Therefore, the dataset’s summary statistics are provided to offer an overview of the variables, each of which plays a role in understanding the energy consumption patterns. The summary statistics of numerical variables are as follows: 

## 4.4 Summary statistics insights of numerical variables

In [20]:
df = pd.read_csv("C:/Users/madsh/OneDrive/Dokumenter/kandidat/Fællesmappe/Forecasting-energy-consumption/Data Cleaning/output_file.csv", encoding="utf-8")

In [19]:
import pandas as pd

# Assume your dataframe is named 'df'
# Selecting the columns of interest for the summary statistics
columns_of_interest = ['Electric cars', 'Plug-in hybrid cars', 'humidity_past1h', 'temp_mean_past1h', 'wind_speed_past1h', 'EL_price']

# Calculating the summary statistics
summary_stats = df[columns_of_interest].describe().transpose()
# Adding the sum and count manually
summary_stats['count'] = df[columns_of_interest].count()

# Renaming columns to match your required format
summary_stats.columns = ['Count', 'Mean', 'Std', 'Min', '25%', '50%', '75%', 'Max']

# Rearranging the order of columns to match your required format
summary_stats = summary_stats[['Count', 'Mean', 'Std', 'Min', '25%', '50%', '75%', 'Max']].transpose()
summary_stats

Electric cars  Plug-in hybrid cars  humidity_past1h  temp_mean_past1h  \
Count   26304.000000         26304.000000     26304.000000      26304.000000   
Mean        0.007992             0.008540        81.429244          8.331720   
Std         0.004053             0.005619        10.658195          6.150023   
Min        -0.012073            -0.027599        39.800000         -9.755000   
25%         0.005086             0.003659        75.193548          3.242623   
50%         0.007145             0.008175        84.360656          8.131967   
75%         0.010764             0.010722        89.606557         13.295020   
Max         0.036147             0.062953        97.934426         26.935593   

       wind_speed_past1h      EL_price  
Count       26304.000000  26304.000000  
Mean            4.834021      0.816396  
Std             1.990381      0.904135  
Min             0.980000     -0.365245  
25%             3.319643      0.219713  
50%             4.551724      0.465135  
75%             6.061507      1.107715  
Max            16.248214      6.478240

Looking at the growth patterns in the table above, the mean percentages for Electric cars and Plug-in hybrid cars indicate a steady growth trend, with means of 0.007992 and 0.008540, respectively. This steady growth, combined with the low standard deviations (0.004053 for Electric cars and 0.005619 for Plug-in hybrid cars), suggests a consistent, albeit modest, increase in new electric and hybrid vehicle registrations. The growth pattern shows a relatively narrow range between the maximum and minimum values: Electric cars ranges from -0.012073 to 0.036147, while Plug-in hybrid cars ranges from -0.027599 to 0.062953. This indicates that the market is expanding at a steady rate during the period of interest, without sudden spikes or drops. The difference in growth rates between electric and hybrid vehicles may reflect varied consumer preferences or policy incentives targeted at specific vehicle types. The slightly higher growth rate for hybrid vehicles might indicate a transitional preference among consumers, moving from conventional gas engines to fully electric options as technology and infrastructure improve.

However, looking at the period shortly after our scope shows the import of electric cars as a percentage change growth rate would be larger than that of the hybrid cars (Elbiler Udgjorde 36 Pct. Af De Nye Biler I 2023, n.d.). The environmental factors (humidity_past1h, temp_mean_past1h, and wind_speed_past1h) show a high degree of stability, with relatively low standard deviations (10.658195 for humidity_past1h, 6.150023 for temp_mean_past1h, and 1.990381 for wind_speed_past1h). This suggests that the observed period did not have significant environmental fluctuations that could have a large effect on energy consumption patterns. The direct impact of these environmental conditions on energy consumption, especially the effect they have on the need for heating or cooling, could be more nuanced. For instance, moderate temperatures might reduce the need for extensive heating or cooling, potentially moderating energy consumption related to climate control for private consumers.

Lastly, the electricity dynamics, consisting of EL_price and energy consumption (Sum_quantity), show minimal fluctuation in electricity price (standard deviation of 0.904135) compared to energy consumption (7.947110). This suggests that while electricity pricing might be regulated or have less volatility due to different means of production, the consumption patterns vary significantly among consumers over time. This variability could reflect different aspects that influence consumer behavior, not directly tied to price changes, such as seasonality and temporal patterns. The dataset includes private customers, who likely have clear usage patterns, using minimal electricity while at work and more when at home. The substantial difference between the 75th percentile (1.107715) and the maximum value (6.478240) for Sum_quantity points to sporadic high consumption events, possibly due to holidays like Christmas when consumers use more electricity-demanding systems simultaneously (Juleaften: Øerne Bruger Mest Ekstra Strøm, Mens Storbykommuner Bruger Mindst, 2022). This is accounted for with the IsHoliday variable, which helps capture these outliers.

Furthermore, the table highlights the different scales of the numerical variables, an aspect that can disrupt machine learning models' prediction accuracy. Therefore, it is evident from the tables that the numerical values should be scaled similarly before applying machine learning models. Analyzing the summary statistics allows a deeper exploration of the dataset's essential aspects: interrelations among variables, aggregating data over customer groups, and time-based dimensions. The next chapter is dedicated to examining these correlations, providing insight into the impact of these dimensions.